integ-test/src/test/java/org/opensearch/sql/sql/PaginationIT.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.sql;
+
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_BEER;
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_CALCS;
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_ONLINE;
+
+import java.io.IOException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.opensearch.sql.legacy.SQLIntegTestCase;
+
+public class PaginationIT extends SQLIntegTestCase {
+  @Override
+  public void init() throws IOException {
+    loadIndex(Index.CALCS);
+    loadIndex(Index.BEER);
+    loadIndex(Index.ONLINE);
+  }
+
+  @Test
+  public void testSmallDataSet() throws IOException {
+    var query = "SELECT * from " + TEST_INDEX_CALCS;
+    var response = new JSONObject(executeFetchQuery(query, 4, "jdbc"));
+    assertTrue(response.has("cursor"));
+    assertEquals(4, response.getInt("total"));
```

Comment:
There is a bug in v2 in pagination, that we have to fix. `total` should be 17 (`= count(*)`), but `size` should be 4 (`= fetch_size`)

Reply:
I think we need a test for these values on all pages